### ISSUE TITLE
Assign a metacell

### ISSUE THREAD

**Issue by annrosebright** [tanaylab/metacell#77]
Hi,

Thank you for this great tool. I wanted to know if I could use metacell in a supervised way. My single-cell RNAseq data also consists of lineage barcodes. What I want to do is to define the cells with certain clones(barcodes) as metacell so that I can define the others (which lack clones) based on that. Is it possible to do that? Instead of identifying the metacell, feed the program with the cell of your choice. 

The reason we want to try this is  @that usually for such experiments only very few cells end up having clones and it makes it hard to perform differential analysis. 

Best,
Ann

**Comment by UboCA** [comment]
@annrosebright
Hi,
 Is it possible to add lineage barcodes into gene expression matrix, and to make it be part of highly variable genes list?
So that dimension reduction process could use your barcode information.

**Comment by aviezerl** [comment]
Yes — `mc@mc` is just a named integer vector mapping cell barcodes to metacell IDs. You can create your own assignment directly:

```r
mc = scdb_mc("your_mc_id")

# Define your own assignments: cells sharing a clone barcode → same metacell
mc@mc = your_custom_named_vector  # names = cell barcodes, values = metacell IDs
```

You'll also need to recompute `mc@mc_fp` (footprint) and `mc@e_gc` (expression) to match your custom grouping, so it's easier to use `mcell_mc_from_coclust_balanced` or similar and then manually override.

For the second part — assigning unbarcoded cells based on your pre-defined metacells — the Python [metacells](https://github.com/tanaylab/metacells) package has a built-in projection pipeline (`mc.pl.projection_pipeline`) designed exactly for this: projecting query cells onto a reference atlas. We recommend using it for this use case.
